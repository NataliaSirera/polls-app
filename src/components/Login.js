import { useState } from "react";
import { useNavigate } from "react-router-dom";
import { connect } from "react-redux";
import { handleLogin } from "../actions/authedUser";

const Login = ({ dispatch }) => {
  const navigate = useNavigate();

  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [logError, setLogError] = useState(false);

  const handleUsername = (e) => {
    const text = e.target.value;
    setUsername(text);
  };

  const handlePassword = (e) => {
    const text = e.target.value;
    setPassword(text);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(handleLogin(username, password))
      .then(() => navigate("/"))
      .catch(() => setLogError(true));
  };

  return (
    <div className="px-8 flex h-screen max-w-screen-md mx-auto">
      <div className="my-auto w-full">
        <h1 className="text-3xl font-bold mb-6 text-center">Employee Polls</h1>
        <h2 className="text-2xl font-medium mb-2 text-center">Log In</h2>
        <form onSubmit={handleSubmit}>
          <div>
            <label className="block mb-2 text-lg">User</label>
            <div>
              <input
                value={username}
                onChange={handleUsername}
                type="text"
                name="username"
                id="username"
                placeholder="User"
                className="px-3 py-2 bg-white border border-gray-300 shadow-sm w-full rounded placeholder-gray-400"
              />
            </div>
          </div>

          <div className="mt-3">
            <label className="block mb-2 text-lg">Password</label>
            <div className="mt-1">
              <input
                value={password}
                onChange={handlePassword}
                type="password"
                name="password"
                id="password"
                placeholder="Password"
                className="px-3 py-2 bg-white border border-gray-300 shadow-sm w-full rounded placeholder-gray-400"
              />
            </div>
          </div>
          {logError && (
            <p className="text-red-700 mt-4">
              The username or password is incorrect. Please try again.
            </p>
          )}
          <div className="mt-6 text-center">
            <button
              type="submit"
              disabled={username === "" || password === ""}
              className="px-5 py-2 text-sm font-medium text-white bg-sky-400 hover:bg-sky-500 rounded-md
            disabled:bg-gray-300 hover:disabled:bg-gray-300"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default connect()(Login);
