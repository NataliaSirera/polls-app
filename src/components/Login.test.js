import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { fireEvent, render } from "@testing-library/react";
import Login from "./Login";
import store from "../store";

describe("Login", () => {
  it("renders the component", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("logs a user in", async () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    const username = component.getByTestId("username");
    const password = component.getByTestId("password");
    fireEvent.change(username, { target: { value: "sarahedo" } });
    fireEvent.change(password, { target: { value: "password123" } });
    expect(username.value).toBe("sarahedo");
    expect(password.value).toBe("password123");
  });
});
