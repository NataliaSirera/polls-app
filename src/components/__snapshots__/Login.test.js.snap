// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="px-8 flex h-screen max-w-screen-md mx-auto"
      >
        <div
          class="my-auto w-full"
        >
          <h1
            class="text-3xl font-bold mb-6 text-center"
          >
            Employee Polls
          </h1>
          <h2
            class="text-2xl font-medium mb-2 text-center"
          >
            Log In
          </h2>
          <form>
            <div>
              <label
                class="block mb-2 text-lg"
              >
                User
              </label>
              <div>
                <input
                  class="px-3 py-2 bg-white border border-gray-300 shadow-sm w-full rounded placeholder-gray-400"
                  data-testid="username"
                  id="username"
                  name="username"
                  placeholder="User"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="mt-3"
            >
              <label
                class="block mb-2 text-lg"
              >
                Password
              </label>
              <div
                class="mt-1"
              >
                <input
                  class="px-3 py-2 bg-white border border-gray-300 shadow-sm w-full rounded placeholder-gray-400"
                  data-testid="password"
                  id="password"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              class="mt-6 mb-6 text-center"
            >
              <button
                class="px-5 py-2 text-sm font-medium text-white bg-sky-400 hover:bg-sky-500 rounded-md disabled:bg-gray-300 hover:disabled:bg-gray-300"
                disabled=""
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="px-8 flex h-screen max-w-screen-md mx-auto"
    >
      <div
        class="my-auto w-full"
      >
        <h1
          class="text-3xl font-bold mb-6 text-center"
        >
          Employee Polls
        </h1>
        <h2
          class="text-2xl font-medium mb-2 text-center"
        >
          Log In
        </h2>
        <form>
          <div>
            <label
              class="block mb-2 text-lg"
            >
              User
            </label>
            <div>
              <input
                class="px-3 py-2 bg-white border border-gray-300 shadow-sm w-full rounded placeholder-gray-400"
                data-testid="username"
                id="username"
                name="username"
                placeholder="User"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="mt-3"
          >
            <label
              class="block mb-2 text-lg"
            >
              Password
            </label>
            <div
              class="mt-1"
            >
              <input
                class="px-3 py-2 bg-white border border-gray-300 shadow-sm w-full rounded placeholder-gray-400"
                data-testid="password"
                id="password"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="mt-6 mb-6 text-center"
          >
            <button
              class="px-5 py-2 text-sm font-medium text-white bg-sky-400 hover:bg-sky-500 rounded-md disabled:bg-gray-300 hover:disabled:bg-gray-300"
              disabled=""
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
